tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,157 @@
+import dataclasses
+from string import Template
+from typing import Optional
+
+import tmt
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.steps.provision.testcloud import GuestTestcloud
+from tmt.utils import field
+
+
+@dataclasses.dataclass
+class BootcData(tmt.steps.provision.testcloud.ProvisionTestcloudData):
+    containerfile: Optional[str] = field(
+        default=None,
+        option=('--containerfile'),
+        metavar='CONTAINERFILE',
+        help="""
+             Select container file to be used to build a container image
+             that is then used by bootc image builder to create a disk image.
+
+             Cannot be used with containerimage.
+             """)
+
+    containerfile_workdir: str = field(
+        default=".",
+        option=('--containerfile-workdir'),
+        metavar='CONTAINERFILE_WORKDIR',
+        help="""
+             Select working directory for the podman build invocation.
+             """)
+
+    containerimage: Optional[str] = field(
+        default=None,
+        option=('--containerimage'),
+        metavar='CONTAINERIMAGE',
+        help="""
+             Select container image to be used to build a bootc disk.
+             This takes priority over containerfile.
+             """)
+
+    add_deps: bool = field(
+        default=True,
+        is_flag=True,
+        option=('--add-deps'),
+        help="""
+             Add tmt dependencies to the supplied container image or image built
+             from the supplied Containerfile.
+             This will cause a derived image to be built from the supplied image.
+             """)
+
+    def set_image(self, image: str) -> None:
+        self.image = image
+
+
+@tmt.steps.provides_method('bootc')
+class ProvisionBootc(tmt.steps.provision.ProvisionPlugin[BootcData]):
+    """
+    Provision guest using bootc
+
+    Minimal configuration using the latest nothing image:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            containerfile: file://Containerfile
+
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+
+    # build a "modified" container image from the base image with tmt dependencies added
+    def _build_derived_image(self, base_image: str) -> str:
```

Comment:
I went this route instead of using podman-bootc (to leverage smbios) primarily because podman-bootc is not yet released in the official repos (it's in [German's copr repo](https://github.com/containers/podman-bootc?tab=readme-ov-file#fedora)).